packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -168,6 +172,12 @@ const ProductTile = (props) => {
         return labelsMap
     }, [product, badgeDetails])
 
+    const shouldShowPromoCallout = (product) => {
```

Comment:
Should we move this function outside of the Component to avoid it being re-defined whenever the tile is re-render?

Reply:
Agree. We can make it as a helper function (`hasPromotions(product)`)  in the product utils file